### PR TITLE
Data cleanup

### PR DESCRIPTION
**Pull request by philip-cc**
### Description
* Moved all data files to a subpackage.
* Removed some unused functions.

### Type(s) of Change (Split the PR if you check many)

- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [x] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Kubernetes changes
- [ ] CI/CD workflow changes

### General Checklist

- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [ ] I have updated relevant Kubernetes manifests/scripts.
- [x] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [ ] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: data/group.go [hunk 1]
```diff
@@ -50,17 +50,17 @@ func (g *group) seal(id uuid.UUID, cryptor crypto.CryptorInterface) (SealedGroup
...
-		return group{}, err
+		return Group{}, err
 	}
 	return plainGroup, nil
 }
 
-// verify checks the integrity of the sealed group.
-func (g *SealedGroup) verify(cryptor crypto.CryptorInterface) bool {
-	_, err := g.unseal(cryptor)
+// Verify checks the integrity of the sealed group.
```
Comment (AmalieDue) [inline data/group.go:62]: Is there a reason why SealedGroup and SealedUser still have a verify method, while all other Sealed-something do not? (There probably is, I just wanted to mention it).

File: data/access_test.go [hunk 2]
```diff
@@ -32,65 +32,65 @@ var accessGroups = []uuid.UUID{
...
 	}
 }
 
 func TestAccessGetZeroGroupIDs(t *testing.T) {
-	access := newAccess(nil)
-	uuids := access.getGroups()
+	access := NewAccess(nil)
+	uuids := access.GetGroups()
 	if len(uuids) != 0 {
 		t.Error("getGroups should have returned empty array")
```
Comment (AmalieDue) [inline data/access_test.go:48]:
```suggestion
		t.Error("GetGroups should have returned empty array")
```

File: data/user_test.go [hunk 3]
```diff
@@ -32,81 +32,81 @@ var userGroups = []uuid.UUID{
...
-	user, _, err := newUser(groupID)
+	user, _, err := NewUser(groupID)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	uuids := user.getGroups()
+	uuids := user.GetGroups()
 	if _, ok := uuids[groupID]; len(uuids) == 0 || !ok {
 		t.Error("Expected getGroups to return a group ID")
```
Comment (AmalieDue) [inline data/user_test.go:42]:
```suggestion
		t.Error("Expected GetGroups to return a group ID")
```

File: data/user_test.go [hunk 4]
```diff
@@ -32,81 +32,81 @@ var userGroups = []uuid.UUID{
...
-	user, _, err := newUser()
+	user, _, err := NewUser()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	uuids := user.getGroups()
+	uuids := user.GetGroups()
 	if len(uuids) != 0 {
 		t.Error("getGroups should have returned empty array")
```
Comment (AmalieDue) [inline data/user_test.go:54]:
```suggestion
		t.Error("GetGroups should have returned empty array")
```

File: data/group.go (hunk 1 above)
Comment (philip-cc) [reply data/group.go:62]: Yes, they are used [here](https://github.com/cyber-crypt-com/encryptonize-lib/blob/1a1ef47a6a5ae043a52b0f644db69ebb42f79a52/encryptonize.go#L100) and [here](https://github.com/cyber-crypt-com/encryptonize-lib/blob/1a1ef47a6a5ae043a52b0f644db69ebb42f79a52/utility.go#L68).

**Review by AmalieDue — APPROVED** [review]

**Pull request merged**